open-metadata-test/open-metadata-fvt/access-services-fvt/subject-area-fvt/src/test/script/configureStartServer.groovy

```diff
@@ -0,0 +1,113 @@
+#!/usr/local/bin/groovy
+import javax.net.ssl.HttpsURLConnection
+
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Contributors to the ODPi Egeria project.
+
+// Function to convert array to String
+
+// Will configure a server chassis - which should already be running - for FVT testing
+
+import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import javax.net.ssl.X509TrustManager
+
+// Retrieve configuration - with defaults to aid in local testing (using default ports)
+user=properties["user"] ?: "garygeeke";
+baseURL=properties["baseURL"] ?: "https://localhost:9443";
+server=properties["server"] ?: "server1";
+retries=properties["retries"] ?: 12;
+delay=properties["delay"] ?: 10;
+
+// SSL setup to avoid self-signed errors for testing
+TrustManager[] trustAllCerts = new TrustManager[]{
+        new X509TrustManager() {
+
+            public java.security.cert.X509Certificate[] getAcceptedIssuers()
+            {
+                return null;
+            }
+            public void checkClientTrusted(java.security.cert.X509Certificate[] certs, String authType)
+            {
+                //No need to implement.
+            }
+            public void checkServerTrusted(java.security.cert.X509Certificate[] certs, String authType)
+            {
+                //No need to implement.
+            }
+        }
+};
+try
+{
+    SSLContext sc = SSLContext.getInstance("SSL");
+    sc.init(null, trustAllCerts, new java.security.SecureRandom());
+    HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
+}
+catch (Exception e)
+{
+    System.out.println(e);
+    System.exit(-1);
+}
+
+// Wait until the platform is ready
+connected=false;
+i=retries;
+while (!connected && i>0)
+{
+    try {
+
+        System.out.println("=== Checking platform at " + baseURL + " is available (" + i + " attempts remaining) ===");
+        post0 = new java.net.URL(baseURL + "/open-metadata/platform-services/users/" + user + "/server-platform/origin").openConnection();
+        post0RC = post0.getResponseCode();
+        println(post0RC);
+        if (post0RC.equals(200)) {
+            connected = true;
+            println(post0.getInputStream().getText());
+        } else {
+            i--;
+            Thread.sleep(1000 * delay);
+        }
+    } catch (Throwable t)
+    {
+        // TODO: look at whether some exceptions should be deemed irrecoverable rather than retry
+        i--;
+        Thread.sleep(1000 * delay);
+    }
+}
+
+// --- Configure the platform - any errors here and we exit
+System.out.println("=== Configuring server: " + server + " ===");
+post1 = new URL(baseURL + "/open-metadata/admin-services/users/" + user + "/servers/" + server + "/local-repository/mode/in-memory-repository" ).openConnection()
```

Comment:
@planetf1 We do have junits that inline the in memory repo to run tests, which means we do not need to have a dependant server. I like being able to run against a real server as you can change the type of repo.  I wonder if we could run the FVTs against an in memory repo and then a graph repo (in case there are quirky differences, for example around search). I raise this as a consideration in case there is something in the current structure you might want to change to accommodate running it against multiple repositories.

Reply:
I did wonder about whether we should run the FVTs twice - once for each repo type (that we ship).

Can you raise a note in the parent issue for that (or a new issue)- as it's just a minor change it will be easy as a new PR? It also means you have a base to work with as you add more tests, or adapt the tests to be more JUnit ish?

As to how that scales.. I think automation is fairly easy, but optimizing performance won't be with all the starts/stops of platforms/servers -- but if we don't do this we may contaminate test results in any case, so probably worth taking the hit

I've also seen v.slow perf recently esp. in graph. I have #3213 but not looked at yet. That is probably a precursor.